src/main/runner.ts

```diff
@@ -190,7 +190,14 @@ export default class RunnerMain extends Runner {
               || this.findForgottenTestById(data.id);
             assert(test, `Couldn't find test by id: ${data.id}`);
 
-            this.emit(event, test, data.err);
+            const testErrors = (test as any).err;
+
+            ((typeof(testErrors) === 'object')
+              && (testErrors !== null)
+                && Object.keys(testErrors).length > 0)
+                  ? this.emit(event, test, Object.assign(testErrors, data.err))
```

Comment:
This seems like a dangerous change to me: testErrors seems to be an object with keys (I'm not sure which keys btw because the type is `any`) and here you assign `data.err` values to `testErrors` thus changing the `testErrors` object. Is that expected?

Reply:
@1999 

Yes, this is expected. I did that because when tests throw an exception (new Error(..)) then:
testErrors is empty:
```
{}
```
And data.err is not:
```
{ message: 'some error',
  name: 'Error',
  stack:
   'Error: some error\n    at Context.<anonymous> (test/parallel-order/tests/parallel2.js:9:14)' }
```

Then when tests assert some result using nodejs assert lib we get:

testErrors:
```
{ generatedMessage: true,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: 'false',
  expected: 'true',
  operator: '==' }
```

data.err
```
{ message:
   'The expression evaluated to a falsy value:\n\n  assert(false)\n',
  name: 'AssertionError [ERR_ASSERTION]',
  stack:
   'AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:\n\n  assert(false)\n\n    at Context.<anonymous> (test/parallel-order/tests/parallel2.js:10:9)' }
```

And when using chai assertion lib:

testErrors
```
{ name: 'AssertionError',
  message: 'expected false to deeply equal true',
  showDiff: true,
  actual: 'false',
  expected: 'true',
  stack:
   'AssertionError: expected false to deeply equal true\n    at Context.<anonymous> (test/parallel-order/tests/parallel2.js:11:26)' }
```

data.err
```
{ message: 'expected false to deeply equal true',
  name: 'AssertionError',
  stack:
   'AssertionError: expected false to deeply equal true\n    at Context.<anonymous> (test/parallel-order/tests/parallel2.js:11:26)' }
```

As far as you can see merging objects will not critically affect emitted err object, but adds necessary fields such as 'actual' and 'expected'.

Of course I could check testErrors for 'actual'/'expected' keys and just add them to data.err. 
What do you think?